test/Test_ArrayComponents.ml

```diff
@@ -0,0 +1,67 @@
+open Base
+open LoopInvGen
+
+
+let select_test () =
+ let arr = Value.Array((Type.INT, Type.INT, [ (Value.Int 3, Value.Int 10)], Value.Int 1))
+ in let select_eval = (Array.of_list ArrayComponents.all).(0).evaluate
+ in let select_ret = (select_eval [arr; (Value.Int 3)])
+ in let res = (Poly.equal select_ret (Value.Int 10))
+ in Alcotest.(check bool) "identical" true res
+
+let select_default_test () =
+ let arr = Value.Array((Type.INT, Type.INT, [ (Value.Int 3, Value.Int 10)], Value.Int 1))
+ in let select_eval = (Array.of_list ArrayComponents.all).(0).evaluate
+ in let select_ret = (select_eval [arr; (Value.Int 4)])
+ in let res = (Poly.equal select_ret (Value.Int 1))
+ in Alcotest.(check bool) "identical" true res
+
+let store_test () =
+ let arr = Value.Array((Type.INT, Type.INT, [ (Value.Int 3, Value.Int 10)], Value.Int 1))
+ in let stored_arr = Value.Array((Type.INT, Type.INT, [ (Value.Int 4, Value.Int 4) ; (Value.Int 3, Value.Int 10)], Value.Int 1))
+ in let store_eval = (Array.of_list ArrayComponents.all).(1).evaluate
+ in let store_ret = (store_eval [arr; (Value.Int 4); (Value.Int 4)])
+ in let res = (Poly.equal stored_arr store_ret)
+ in Alcotest.(check bool) "identical" true res
+
+let select_sem_test() = 
```

Comment:
Minor: use a *space* between function name and the (unit) argument `()` for this and the next 3 functions.